src/react/hooks/auth.ts

```diff
@@ -0,0 +1,40 @@
+import { QueryHookOptions, useQuery } from "@apollo/client";
+import { useContext } from "react";
+import { AuthSDK } from "../../core/auth";
+import { SaleorContext } from "../context";
+import { getUserDetailsQuery } from "../../apollo/queries";
+
+export const useAuth = (): AuthSDK => {
+  const saleor = useContext(SaleorContext);
+
+  if (!saleor) {
+    throw new Error(
+      "Could not find saleor's apollo client in the context. Did you forget to wrap the root component in a <SaleorProvider>?"
+    );
+  }
+
+  return saleor.api.auth;
+};
+
+export const useAuthState = (options?: QueryHookOptions<any, never>) => {
+  const saleor = useContext(SaleorContext);
+
+  if (!saleor) {
```

Comment:
I'm just thinking if this error handler shouldn't be used only in `useAuth` hook. If not we can think how not to duplicate this error handler in every `useContext(SaleorContext);` usage.